src/Provider/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { IComponentMapping } from '@stoplight/markdown-viewer';
+import axios from 'axios';
+import * as React from 'react';
+import { NodeIconMapping } from '../types';
+
+export interface IProvider {
+  host?: string;
+  token?: string;
+  components?: IComponentMapping;
+  icons?: NodeIconMapping;
+}
+
+const defaultHost = 'http://localhost:4060';
+export const HostContext = React.createContext(defaultHost);
+export const AxiosContext = React.createContext(axios.create());
+export const ComponentsContext = React.createContext<IComponentMapping | undefined>(undefined);
+
+const defaultIcons: NodeIconMapping = {};
+export const IconsContext = React.createContext<NodeIconMapping>(defaultIcons);
+
+export const Provider: React.FunctionComponent<IProvider> = ({ host, token, components, icons, children }) => {
```

Comment:
We can get around not moving Provider over so I think it's save to remove. We can just add support for passing a row renderer into the TableOfContents component